new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/TestUtils.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.utils;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.HashMap;
+import java.util.List;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+
+/**
+ * The utility class for tests.
+ */
+public class TestUtils {
+
+  /**
+   * Call the curl command and check the app can be reached from all managed servers.
+   *
+   * @param curlCmd curl command to call the sample app
+   * @param managedServerNames managed server names that the sample app response should return
+   * @param maxIterations max iterations to call the curl command
+   * @return true if the web app can hit all managed servers, false otherwise
+   */
+  public static boolean callWebAppAndCheckForServerNameInResponse(
+                          String curlCmd,
+                          List<String> managedServerNames,
+                          int maxIterations) {
+
+    // first map all server names with false
+    HashMap<String, Boolean> managedServers = new HashMap<>();
+    managedServerNames.forEach(managedServerName ->
+        managedServers.put(managedServerName, false)
+    );
+
+    logger.info("Calling webapp at most {0} times using command: {1}", maxIterations, curlCmd);
+
+    // check the response contains managed server name
+    for (int i = 1; i <= maxIterations; i++) {
+
+      if (!managedServers.containsValue(false)) {
+        return true;
+      } else {
+        try {
+          // sometimes the pod is not ready even the condition check is ready, sleep a little bit
+          Thread.sleep(100);
+          ExecResult result = ExecCommand.exec(curlCmd, true);
+
+          String response = result.stdout().trim();
+          managedServers.keySet().forEach(key -> {
+            if (response.contains(key)) {
+              managedServers.put(key, true);
+            }
+          });
+        } catch (Exception e) {
+          logger.info("Got exceptions while running command: " + curlCmd);
+          return false;
+        }
+      }
+    }
+
+    // after the max iterations, check if any managedserver value is false
+    managedServers.forEach((key, value) -> {
+      if (value) {
+        logger.info("The sample app can be accessed from the server {0}", key);
+      } else {
+        logger.info("FAILURE: The sample app can not be accessed from the server {0}", key);
+      }
+    });
+
+    // final check if any managed server value is false
+    return !managedServers.containsValue(false);
```

Comment:
feels odd to do the same calculation over and over
why not do it once and store the result in a var

Reply:
fixed